### ISSUE TITLE
How does the npy_feature_maps be generated?

### ISSUE THREAD

**Issue by aruiplex** [ThalesGroup/pim-generalized-category-discovery#7]
Hi, I am following up on your work, but could you be more specific about how your `npy_feature_maps.zip` is generated?

When I use the GCD code you mentioned in README.md, it generates the feature maps like this:

```
├── vit_dino_aircraft
│   ├── test
        ├── 0
        ├── 1
        ├── 10
        ├── 100
...
│   └── train
        ├── 0
        ├── 1
        ├── 10
        ├── 100
...
```

It is not similar to your provided feature map structure:

```
└── vit_dino_scars_best
    ├── ssKM_ctrds.pt
    └── train
        ├── all_feats.npy
        ├── mask_cls.npy
        ├── mask_lab.npy
        └── targets.npy
```

Could you provide the method of how you generate this?

**Comment by aruiplex** [comment]
Hi, could you explain this? Very thanks.

**Comment by tds-fchiaroni** [comment]
Hello,

We do not provide the preprocessing code for obtaining the feature maps because our code focuses solely on the exploitation of the feature maps, not their generation.

The contents of the `vit_dino_scars_best` directory are structured differently from the `vit_dino_aircraft` directory that you've described. Here’s a breakdown of the `vit_dino_scars_best` directory:

- `ssKM_ctrds.pt`: 
    - This is a PyTorch file containing the cluster centroids estimated using semi-supervised k-means. 
  
- `train` directory: 
    - This directory contains several files associated with the training set, organized as follows:
    
    - `all_feats.npy`: 
        - A 2D NumPy array with shape `(N, fm)`, where `N` is the number of samples, and `fm` is the dimensionality of the feature maps. This file contains all the feature maps generated for the training set.
    
    - `mask_cls.npy`: 
        - A 1D NumPy array of size `N` used to indicate whether a sample belongs to an old or new class.
            
    - `mask_lab.npy`: 
        - A 1D NumPy array of size `N` that defines which samples are labeled and which are unlabeled.
    
    - `targets.npy`: 
        - A 1D NumPy array of size `N` containing the ground truth class labels for each sample in the training set.

I hope this explanation clarifies the contents. If you have any further questions, feel free to ask!

Best regards.

**Comment by aruiplex** [comment]
Very thanks for your explanation, I already figure out how to generate them by the GCD code. For everyone who wants to follow this work, all these npy files are generated by the [code](https://github.com/sgvaze/generalized-category-discovery/blob/main/methods/clustering/k_means.py), and you need to save the variables to file. 

I don't think there's anything to hide, just make life easier.

**Comment by tds-fchiaroni** [comment]
Thank you for your feedback! We are glad you were able to figure out the process. I want to assure you that transparency is important to us, and we purposefully provided the feature maps to make it easier for users to apply our method, without the need to re-train the entire feature extractor from the GCD code (which is not our contribution). Our contribution focuses specifically on working on top of feature maps, which is why we chose to provide them directly.

We appreciate your suggestion and will certainly keep this in mind as we continue to develop our work. If you need further assistance, please don't hesitate to reach out.

Best regards.
